src/index.js

```diff
@@ -20,6 +20,9 @@ function serializeComponents(props) {
     } else if (props[component].constructor === Object) {
       // Stringify components passed as object.
       serialProps[component] = styleParser.stringify(props[component]);
+    } else if (props[component] === true) {
```

Comment:
I see. In that case, I will determine if the component exists and then set
the empty prop to the default of that component's property, unless "true"
is in the property's oneOf, in which case I will set it as such.

Which will lead me to stringify boolean false as "false" as well.

On Wed, Mar 2, 2016 at 11:11 PM, Kevin Ngo <notifications@github.com> wrote:

> In src/index.js
> <https://github.com/ngokevin/aframe-react/pull/23#discussion_r54839711>:
>
> > @@ -20,6 +20,9 @@ function serializeComponents(props) {
> >      } else if (props[component].constructor === Object) {
> >        // Stringify components passed as object.
> >        serialProps[component] = styleParser.stringify(props[component]);
> > +    } else if (props[component] === true) {
>
> hmm what happens if there's a component that accepts a boolean value that
> defaults to false.
>
> something like visible={true}, which would still work in this case since
> it defaults to true, but it's possible for a component to accept bool
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ngokevin/aframe-react/pull/23/files#r54839711>.
>